src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Reference.cs

```diff
@@ -266,18 +266,35 @@ public void LoadXml(XmlElement value)
                         // let the transform read the children of the transformElement for data
                         transform.LoadInnerXml(transformElement.ChildNodes);
                         // Hack! this is done to get around the lack of here() function support in XPath
-                        if (transform is XmlDsigEnvelopedSignatureTransform
-                            && _uri != null && (_uri.Length == 0 || _uri[0] == '#'))
+                        if (transform is XmlDsigEnvelopedSignatureTransform)
                         {
                             // Walk back to the Signature tag. Find the nearest signature ancestor
                             // Signature-->SignedInfo-->Reference-->Transforms-->Transform
                             XmlNode? signatureTag = transformElement.SelectSingleNode("ancestor::ds:Signature[1]", nsm);
 
                             // Resolve the reference to get starting point for position calculation.
-                            XmlNode? referenceTarget =
-                                _uri.Length == 0
-                                ? transformElement.OwnerDocument
-                                : SignedXml!.GetIdElement(transformElement.OwnerDocument, Utils.GetIdFromLocalUri(_uri, out bool _));
+                            // This needs to match the way CalculateSignature resolves URI references.
+                            XmlNode? referenceTarget = null;
+                            if (_uri == null)
+                            {
+                                referenceTarget = transformElement.OwnerDocument;
+                            }
+                            else if (_uri.Length == 0)
+                            {
+                                referenceTarget = transformElement.OwnerDocument;
+                            }
+                            else if (_uri[0] == '#')
+                            {
+                                var idref = Utils.ExtractIdFromLocalUri(_uri);
+                                if (idref == "xpointer(/)")
```

Comment:
This fixes the very specific case of `#xpointer(/)` but not something like `#xpointer(id("foo"))`. Did that work before? Or only the root document?

Reply:
Yes, that worked correctly before, since `Get/ExtractIdFromLocalUri` would return the `foo` from `xpointer(id("foo"))`, which SignedXml.GetIdElement knows how to handle. `SignedXml.GetIdElement` doesn't know how to handle  `xpointer(/)` and would return `null` in that case, which I think was the root cause for the bug.

Reply:
The `xpointer(id("xyz"))` cases are not tested at all, though. They should behave the same as URI="#id", so it would probably make more sense to tweak the existing tests with such references, to also have the `xpointer(id("id"))` counterparts.